packages/untp-test-suite/src/core/processTestSuite.ts

```diff
@@ -1,19 +1,41 @@
 import {
-  ConfigContent,
-  ConfigCredentials,
+  IConfigContent,
+  ICredentialConfigs,
+  ICredentialTestResult,
   IValidatedCredentials,
-  TestSuite,
+  ITestSuiteResult,
 } from './types/index.js';
 import { readJsonFile, validateCredentialConfigs } from './utils/common.js';
 import { dynamicLoadingSchemaService } from './services/dynamic-loading-schemas/loadingSchema.service.js';
 import { hasErrors } from './services/json-schema/validator.service.js';
-import {  getCredentialResults, getFinalReport } from '../templates/utils.js';
+import {
+  constructCredentialTestResult,
+  constructCredentialTestResults,
+  constructFinalReport,
+} from '../templates/utils.js';
 
-export const processCheckDataBySchema = async (credentialConfig: ConfigContent): Promise<IValidatedCredentials> => {
+export const processCheckDataBySchema = async (
+  credentialConfig: IConfigContent,
+  data?: object,
+): Promise<IValidatedCredentials> => {
   const { type, version, dataPath } = credentialConfig;
-  const [schema, data] = await Promise.all([dynamicLoadingSchemaService(type, version), readJsonFile(dataPath)]);
+  let _data;
+
+  if (!data && !dataPath) {
```

Comment:
I think we should check data is {}. Because when I test with data is {} and without dataPath, the test case still shows a pass

Reply:
Your test schema does not require any properties, so it will pass when test the empty object